scrapy/core/downloader/handlers/s3.py

```diff
@@ -24,21 +24,29 @@ def __init__(self, settings, *,
         # If no credentials could be found anywhere,
         # consider this an anonymous connection request by default;
         # unless 'anon' was set explicitly (True/False).
-        anon = kw.get('anon')
-        if anon is None and not aws_access_key_id and not aws_secret_access_key:
-            kw['anon'] = True
-        self.anon = kw.get('anon')
+        self.anon = kw.pop('anon', None)
+        if kw:
+            raise TypeError(f'Unexpected keyword arguments: {kw}')
 
         self._signer = None
         import botocore.auth
-        import botocore.credentials
-        kw.pop('anon', None)
-        if kw:
-            raise TypeError(f'Unexpected keyword arguments: {kw}')
-        if not self.anon:
-            SignerCls = botocore.auth.AUTH_TYPE_MAPS['s3']
-            self._signer = SignerCls(botocore.credentials.Credentials(
-                aws_access_key_id, aws_secret_access_key))
+        from botocore.credentials import Credentials
+        from botocore.session import Session
+        if self.anon is None or self.anon is False:
+            # Cause Session.create_client() is used in pipelines and feed exporter,
+            # following the credential creation logic in Session.create_client()
+            # maximize consistence of AWS auth across the project.
+            if aws_access_key_id is not None and aws_secret_access_key is not None:
+                credentials = Credentials(access_key=aws_access_key_id,
+                                          secret_key=aws_secret_access_key)
+            else:
+                credentials = Session().get_credentials()
```

Comment:
If this is replicating code somewhere else, any chance we can move shared code into `utils`?

Reply:
The "somewhere else" is in `botocore`'s `Session.create_client()` (only the creadential creation part), not in Scrapy. Considering the snippet is used only once, there's no need to abstract it as a `utils` function.

Reply:
Could we then remove the comment? It seems more like a comment for code review than a comment for future reference.

Reply:
The comment explains why there's two branches used to create a credential. I prefer to keep it to make the code more understandable.

Reply:
I see the comment is about https://github.com/boto/botocore/blob/develop/botocore/session.py#L827-L839. Why don’t we use `Session.create_client()` here directly, as we do in the other parts references in the code comment, instead of replicating part of its implementation manually?